prover-service/src/training_wheels/mod.rs

```diff
@@ -1,8 +1,93 @@
 mod sign;
 pub mod verification_logic;
 
+use crate::api::RequestInput;
+use crate::config::ProverServiceConfig;
+use crate::input_processing::types::VerifiedInput;
+use crate::jwk_fetching;
+use crate::jwk_fetching::get_federated_jwk;
+use crate::state::ProverServiceState;
+use crate::training_wheels::verification_logic::compute_nonce;
+use anyhow::{anyhow, bail, ensure};
+use aptos_keyless_common::input_processing::encoding::{AsFr, DecodedJWT};
+use aptos_keyless_common::logging;
+use aptos_keyless_common::logging::HasLoggableError;
+use aptos_types::jwks::rsa::RSA_JWK;
 pub use sign::sign;
 pub use sign::verify;
-pub use verification_logic::check_nonce_consistency;
-pub use verification_logic::validate_jwt_payload_parsing;
-pub use verification_logic::validate_jwt_sig_and_dates;
+use std::sync::Arc;
+use std::time::{SystemTime, UNIX_EPOCH};
+pub use verification_logic::validate_jwt_sig;
+
+/// All training-wheel checks go here.
+/// If a request passes this, we should be convinced that this is the *public statement* to be proved is correct.
+pub async fn preprocess_and_validate_request(
+    prover: &ProverServiceState,
+    req: &RequestInput,
+) -> anyhow::Result<VerifiedInput> {
+    let _span = logging::new_span("TrainingWheelChecks");
+    let jwt = DecodedJWT::from_b64(&req.jwt_b64).log_err()?;
+    let jwk = finalize_jwk(&prover.config, &jwt).await.log_err()?;
+
+    {
+        let _span = logging::new_span("VerifyJWTSignature");
+        validate_jwt_sig(jwk.as_ref(), &req.jwt_b64, &prover.config).log_err()?;
+    }
+
+    if prover.config.enable_jwt_iat_not_in_future_check {
```

Comment:
Can you please 🙏 clarify when this would ever be disabled? And can you please add it as a comment above this?

Reply:
this is moved from [here](https://github.com/aptos-labs/keyless-zk-proofs/pull/68/files#r2196125051), not sure why the original author did this.
should it be a separate investigation?

Reply:
Leave a TODO to understand when this would ever be disabled please 🙏!

Reply:
added todo